ocs_ci/ocs/platform_nodes.py

```diff
@@ -217,14 +217,25 @@ def get_vms(self, nodes):
             list: vSphere vm objects list
 
         """
-        vms_in_pool = self.vsphere.get_all_vms_in_pool(
-            self.cluster_name, self.datacenter, self.cluster
-        )
-        node_names = [node.get().get("metadata").get("name") for node in nodes]
+        vms_all = []
         vms = []
+        logger.info(f"deployment type: {self.deployment_type}")
+        if self.deployment_type == "upi":
+            vms_all = self.vsphere.get_all_vms_in_pool(
+                self.cluster_name, self.datacenter, self.cluster
+            )
+        elif self.deployment_type == "ipi":
+            vms_all = self.vsphere.get_all_vms_in_dc(self.datacenter)
```

Comment:
Wouldn't it mean that if you are getting all the VMs in the datacenter, you will get not only the VMs in the relevant ODF cluster but all other VMs in the DC, of all other ODF clusters running in that DC?

Reply:
yes, but then I filter the once I need by the `nodes` variable passed to the function

Reply:
oh I see. Thanks for clarifting